dotcom-rendering/src/components/Picture.tsx

```diff
@@ -318,7 +319,22 @@ export const Picture = ({
 	isLazy = true,
 	isLightbox = false,
 	orientation = 'landscape',
+	onLoad,
 }: Props) => {
+	const ref = useRef<HTMLImageElement>(null);
+	const [loaded, setLoaded] = useState(false);
+
+	useEffect(() => {
+		if (!ref.current) return;
+
+		if (ref.current.complete) return setLoaded(true);
+		ref.current.addEventListener('load', () => setLoaded(true));
+	}, [ref]);
```

Comment:
I don't think passing a ref to a useEffect will work as expected as the ref object will have the same referential equality across renders so it won't trigger the useEffect if it changes. What you might want instead is a useCallback ref:

https://medium.com/welldone-software/usecallback-might-be-what-you-meant-by-useref-useeffect-773bc0278ae